src/audio/dai.c

```diff
@@ -800,6 +802,15 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 		dd->config.burst_elems =
 			dd->dai->plat_data.fifo[dev->params.direction].depth;
 		break;
+	case SOF_DAI_IMX_ESAI:
+		channel = EDMA_HS_GET_CHAN(handshake);
+		trace_dai_with_ids(dev, "dai_config() has done ESAI specific channel selection");
+		dd->frame_bytes = 4; /* The ESAI works with 24 bit samples, padded to 32 bits */
+		if (dev->params.frame_fmt == SOF_IPC_FRAME_S16_LE)
+			dd->frame_bytes = 2; /* The other formats have 4 bytes */
```

Comment:
I think if ... else ... is more understandable at first glance, but it's up to you.

Reply:
Changing it to a switch statement.